### ISSUE TITLE
The expected results are not achieved on the tiered-imagenet

### ISSUE THREAD

**Issue by 1156477988** [yinboc/few-shot-meta-baseline#35]
Thank you for sharing, I really appreciate it. I recently used your code to reproduce the experimental results in your paper, and I had no problem with the results on mini-imagenet, but when I used the tiered-resnet12.pth file from the provided pre-trained model to train on the tiered-imagenet dataset, I found that with the 5way-1shot setting, I could only achieve about 63% and could not achieve 68% of the results in the paper.

**Comment by yinboc** [comment]
The experiments should still be reproducible with latest pytorch. 63% sounds like the accuracy on validation set, what numbers did you get in "tval: xxx" from logs when running meta-learning?

**Comment by 1156477988** [comment]
train dataset: torch.Size([3, 84, 84]) (x448695), 351
tval dataset: torch.Size([3, 84, 84]) (x206209), 160
val dataset: torch.Size([3, 84, 84]) (x124261), 97
num params: 8.0M
epoch 1, train 0.4949|0.8945, tval 0.7002|0.7824, val 0.8039|0.7419, 4.7m 4.7m/1.6h (@24)
epoch 2, train 0.4292|0.8927, tval 0.6820|0.7782, val 0.7776|0.7445, 4.8m 9.5m/1.6h (@24)
epoch 3, train 0.4168|0.8930, tval 0.6790|0.7782, val 0.7696|0.7429, 4.8m 14.3m/1.6h (@24)
epoch 4, train 0.3966|0.8970, tval 0.6651|0.7785, val 0.7579|0.7453, 5.0m 19.3m/1.6h (@24)
epoch 5, train 0.3965|0.8946, tval 0.6719|0.7768, val 0.7600|0.7451, 4.8m 24.1m/1.6h (@24)
epoch 6, train 0.3857|0.8979, tval 0.6614|0.7818, val 0.7434|0.7467, 4.8m 29.0m/1.6h (@24)
epoch 7, train 0.3681|0.9014, tval 0.6576|0.7800, val 0.7437|0.7457, 4.9m 33.9m/1.6h (@24)
epoch 8, train 0.3743|0.8977, tval 0.6567|0.7773, val 0.7370|0.7463, 4.9m 38.7m/1.6h (@24)
epoch 9, train 0.3669|0.8991, tval 0.6550|0.7772, val 0.7361|0.7452, 4.9m 43.6m/1.6h (@24)
epoch 10, train 0.3558|0.9028, tval 0.6466|0.7787, val 0.7316|0.7445, 4.9m 48.5m/1.6h (@24)
epoch 11, train 0.3478|0.9037, tval 0.6531|0.7777, val 0.7305|0.7469, 4.9m 53.4m/1.6h (@24)
epoch 12, train 0.3548|0.9012, tval 0.6528|0.7763, val 0.7315|0.7447, 4.9m 58.3m/1.6h (@24)
epoch 13, train 0.3465|0.9051, tval 0.6485|0.7772, val 0.7233|0.7461, 4.9m 1.1h/1.6h (@24)
epoch 14, train 0.3391|0.9049, tval 0.6443|0.7768, val 0.7267|0.7441, 4.9m 1.1h/1.6h (@24)
epoch 15, train 0.3324|0.9079, tval 0.6442|0.7750, val 0.7225|0.7469, 4.9m 1.2h/1.6h (@24)
epoch 16, train 0.3296|0.9056, tval 0.6489|0.7763, val 0.7223|0.7453, 4.9m 1.3h/1.6h (@24)
epoch 17, train 0.3279|0.9073, tval 0.6489|0.7751, val 0.7266|0.7405, 4.9m 1.4h/1.6h (@24)
epoch 18, train 0.3153|0.9092, tval 0.6536|0.7706, val 0.7248|0.7428, 4.9m 1.5h/1.6h (@24)
epoch 19, train 0.3171|0.9107, tval 0.6460|0.7748, val 0.7258|0.7428, 4.9m 1.5h/1.6h (@24)
epoch 20, train 0.3188|0.9071, tval 0.6416|0.7767, val 0.7163|0.7432, 4.9m 1.6h/1.6h (@24)


This is the result of the 5-way-5-shot experiment on tiered-imagenet, using the pre-trained model provided, with all parameters set in the original code.This did not achieve the expected result.

**Comment by 1156477988** [comment]
@yinboc 
Thank you for your reply and here are some of my experimental results:

train dataset: torch.Size([3, 84, 84]) (x448695), 351
tval dataset: torch.Size([3, 84, 84]) (x206209), 160
val dataset: torch.Size([3, 84, 84]) (x124261), 97
num params: 8.0M
epoch 1, train 0.4949|0.8945, tval 0.7002|0.7824, val 0.8039|0.7419, 4.7m 4.7m/1.6h (@24)
epoch 2, train 0.4292|0.8927, tval 0.6820|0.7782, val 0.7776|0.7445, 4.8m 9.5m/1.6h (@24)
epoch 3, train 0.4168|0.8930, tval 0.6790|0.7782, val 0.7696|0.7429, 4.8m 14.3m/1.6h (@24)
epoch 4, train 0.3966|0.8970, tval 0.6651|0.7785, val 0.7579|0.7453, 5.0m 19.3m/1.6h (@24)
epoch 5, train 0.3965|0.8946, tval 0.6719|0.7768, val 0.7600|0.7451, 4.8m 24.1m/1.6h (@24)
epoch 6, train 0.3857|0.8979, tval 0.6614|0.7818, val 0.7434|0.7467, 4.8m 29.0m/1.6h (@24)
epoch 7, train 0.3681|0.9014, tval 0.6576|0.7800, val 0.7437|0.7457, 4.9m 33.9m/1.6h (@24)
epoch 8, train 0.3743|0.8977, tval 0.6567|0.7773, val 0.7370|0.7463, 4.9m 38.7m/1.6h (@24)
epoch 9, train 0.3669|0.8991, tval 0.6550|0.7772, val 0.7361|0.7452, 4.9m 43.6m/1.6h (@24)
epoch 10, train 0.3558|0.9028, tval 0.6466|0.7787, val 0.7316|0.7445, 4.9m 48.5m/1.6h (@24)
epoch 11, train 0.3478|0.9037, tval 0.6531|0.7777, val 0.7305|0.7469, 4.9m 53.4m/1.6h (@24)
epoch 12, train 0.3548|0.9012, tval 0.6528|0.7763, val 0.7315|0.7447, 4.9m 58.3m/1.6h (@24)
epoch 13, train 0.3465|0.9051, tval 0.6485|0.7772, val 0.7233|0.7461, 4.9m 1.1h/1.6h (@24)
epoch 14, train 0.3391|0.9049, tval 0.6443|0.7768, val 0.7267|0.7441, 4.9m 1.1h/1.6h (@24)
epoch 15, train 0.3324|0.9079, tval 0.6442|0.7750, val 0.7225|0.7469, 4.9m 1.2h/1.6h (@24)
epoch 16, train 0.3296|0.9056, tval 0.6489|0.7763, val 0.7223|0.7453, 4.9m 1.3h/1.6h (@24)
epoch 17, train 0.3279|0.9073, tval 0.6489|0.7751, val 0.7266|0.7405, 4.9m 1.4h/1.6h (@24)
epoch 18, train 0.3153|0.9092, tval 0.6536|0.7706, val 0.7248|0.7428, 4.9m 1.5h/1.6h (@24)
epoch 19, train 0.3171|0.9107, tval 0.6460|0.7748, val 0.7258|0.7428, 4.9m 1.5h/1.6h (@24)
epoch 20, train 0.3188|0.9071, tval 0.6416|0.7767, val 0.7163|0.7432, 4.9m 1.6h/1.6h (@24)

This is the result of the 5-way-5-shot experiment on tiered-imagenet, using the pre-trained model provided, with all parameters set in the original code.This did not achieve the expected result.

**Comment by yinboc** [comment]
It seems you modified the code. "train dataset: torch.Size([3, 84, 84]) (x448695), 351" is not expected, the original code should have "torch.Size([3, 80, 80])", i.e. the input image size is 80x80.

If you want to train with 84x84, it should still work, but in this case you should not use the pre-trained models since they are trained for 80x80 with the original code.

**Comment by 1156477988** [comment]
After resizing the image to 3, 80, 80, I still did not achieve the results given in the paper.


set gpu: 0
train dataset: torch.Size([3, 80, 80]) (x448695), 351
tval dataset: torch.Size([3, 80, 80]) (x206209), 160
val dataset: torch.Size([3, 80, 80]) (x124261), 97
classifier
num params: 8.0M
epoch 1, train 0.4775|0.9005, tval 0.6930|0.7861, val 0.7922|0.7471, 3.1m 3.1m/1.0h (@24)                                                 
epoch 2, train 0.4167|0.8981, tval 0.6762|0.7840, val 0.7688|0.7486, 3.1m 6.2m/1.0h (@24)                                                 
epoch 3, train 0.4070|0.8970, tval 0.6696|0.7853, val 0.7617|0.7478, 3.1m 9.3m/1.0h (@24)

**Comment by 1156477988** [comment]
@yinboc 

After resizing the image to 3, 80, 80, I still did not achieve the results given in the paper.
This model generally converges very quickly, and the later epochs do not change much, so the overall conclusion can be seen from the first three results.

set gpu: 0
train dataset: torch.Size([3, 80, 80]) (x448695), 351
tval dataset: torch.Size([3, 80, 80]) (x206209), 160
val dataset: torch.Size([3, 80, 80]) (x124261), 97
classifier
num params: 8.0M
epoch 1, train 0.4775|0.9005, tval 0.6930|0.7861, val 0.7922|0.7471, 3.1m 3.1m/1.0h (@24)
epoch 2, train 0.4167|0.8981, tval 0.6762|0.7840, val 0.7688|0.7486, 3.1m 6.2m/1.0h (@24)
epoch 3, train 0.4070|0.8970, tval 0.6696|0.7853, val 0.7617|0.7478, 3.1m 9.3m/1.0h (@24)

**Comment by 1156477988** [comment]
Also, when using the mini-imagenet dataset, the results are similar when loading the pre-trained model for meta-train with image sizes of 84,84 and 80,80.

**Comment by 1156477988** [comment]
@yinboc 

train dataset: torch.Size([3, 80, 80]) (x448695), 351
tval dataset: torch.Size([3, 80, 80]) (x206209), 160
val dataset: torch.Size([3, 80, 80]) (x124261), 97
num params: 8.0M
epoch 1, train 0.4775|0.9005, tval 0.6930|0.7861, val 0.7922|0.7471, 3.1m 3.1m/1.0h (@24)
epoch 2, train 0.4167|0.8981, tval 0.6762|0.7840, val 0.7688|0.7486, 3.1m 6.2m/1.0h (@24)
epoch 3, train 0.4070|0.8970, tval 0.6696|0.7853, val 0.7617|0.7478, 3.1m 9.3m/1.0h (@24)
epoch 4, train 0.3878|0.9001, tval 0.6623|0.7794, val 0.7539|0.7494, 3.1m 12.4m/1.0h (@24)
epoch 5, train 0.3886|0.8984, tval 0.6665|0.7819, val 0.7564|0.7488, 3.1m 15.5m/1.0h (@24)
epoch 6, train 0.3773|0.9007, tval 0.6569|0.7828, val 0.7376|0.7514, 3.1m 18.6m/1.0h (@24)
epoch 7, train 0.3635|0.9033, tval 0.6517|0.7838, val 0.7395|0.7496, 3.1m 21.7m/1.0h (@24)
epoch 8, train 0.3664|0.9019, tval 0.6453|0.7842, val 0.7323|0.7478, 3.1m 24.8m/1.0h (@24)
epoch 9, train 0.3596|0.9020, tval 0.6521|0.7803, val 0.7299|0.7498, 3.1m 27.9m/1.0h (@24)
epoch 10, train 0.3521|0.9051, tval 0.6442|0.7792, val 0.7234|0.7482, 3.1m 31.0m/1.0h (@24)
epoch 11, train 0.3407|0.9068, tval 0.6452|0.7828, val 0.7231|0.7486, 3.1m 34.1m/1.0h (@24)
epoch 12, train 0.3514|0.9037, tval 0.6479|0.7791, val 0.7294|0.7445, 3.1m 37.2m/1.0h (@24)
epoch 13, train 0.3424|0.9062, tval 0.6434|0.7800, val 0.7231|0.7493, 3.1m 40.4m/1.0h (@24)
epoch 14, train 0.3372|0.9063, tval 0.6390|0.7800, val 0.7224|0.7483, 3.1m 43.5m/1.0h (@24)
epoch 15, train 0.3281|0.9092, tval 0.6420|0.7756, val 0.7169|0.7443, 3.1m 46.6m/1.0h (@24)
epoch 16, train 0.3261|0.9085, tval 0.6448|0.7762, val 0.7166|0.7478, 3.1m 49.7m/1.0h (@24)
epoch 17, train 0.3255|0.9087, tval 0.6433|0.7789, val 0.7156|0.7463, 3.1m 52.8m/1.0h (@24)
epoch 18, train 0.3105|0.9133, tval 0.6377|0.7776, val 0.7170|0.7447, 3.1m 55.9m/1.0h (@24)
epoch 19, train 0.3140|0.9125, tval 0.6381|0.7781, val 0.7172|0.7484, 3.1m 59.0m/1.0h (@24)
epoch 20, train 0.3172|0.9090, tval 0.6362|0.7802, val 0.7112|0.7462, 3.1m 1.0h/1.0h (@24)

**Comment by yinboc** [comment]
It seems I can reproduce the correct results on both datasets. I am not sure what issues are there in your case, I feel it might be helpful to:

1. make sure to use the orignal code and carefully follow the instructions
2. make sure you downloaded the correct pre-trained model
3. also evaluate mini-imagenet, or Classifier-Baseline only, to see whether this issue is specific for meta-tiered

Hopefully you may solve this issue.

**Comment by AnCyanQ** [comment]
> 
Hi, I'm also confused by the same problem, could you share your data enhancement method?
Thanks in advance!

**Comment by zhuolingli** [comment]
> The problem may have been solved, which may be the data enhancement method in the **tiered Imagenet** dataset. Thank you for your answers。

Hello, I am also confused by the same question, can you share your data enhancement method?
Thanks in advance!
